src/closed.jl

```diff
@@ -0,0 +1,54 @@
+"""
+Defines the ClosedSet type.
+
+A ClosedSet(left, right) is an interval set where both bounds are closed i.e. the range is [left, right]
+"""
+immutable ClosedSet{T}
+    left::T
+    right::T
+end
+
+function ClosedSet(left, right)
```

Comment:
This can be a one-liner, if you wish. `@code_llvm` suggests that the 2-argument case does not suffer from splatting penalty (perhaps it's special-cased in the compiler).